lib/application.js

```diff
@@ -301,18 +301,16 @@ app.enableAuth = function() {
     var Model = method.ctor;
     var modelInstance = ctx.instance;
 
-    var modelId = modelInstance && modelInstance.id ||
-      // replacement for deprecated req.param()
-      (req.params && req.params.id !== undefined ? req.params.id :
-       req.body && req.body.id !== undefined ? req.body.id :
-       req.query && req.query.id !== undefined ? req.query.id :
-       undefined);
+    var modelId = modelInstance && modelInstance.id || ctx.args.id;
+    ctx.modelId = modelId;
 
     var modelName = Model.modelName;
 
     var modelSettings = Model.settings || {};
     var errStatusCode = modelSettings.aclErrorStatus || app.get('aclErrorStatus') || 401;
-    if (!req.accessToken) {
+    if (req.accessToken) {
+      ctx.accessToken = req.accessToken;
```

Comment:
Well, is it necessary for the auth hook to share the `ctx` object with strong-remoting? I mean you don't have to pass strong-remoting context to the auth hook, in my opinion you are free to create your own auth-hook-specific context.